sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py

```diff
@@ -648,15 +652,34 @@ def run_joint_trajectory_unsafe(self, joint_trajectory, wait=True):
         trajectory of the joints which would be executed.
         @param wait - should method wait for movement end or not
         """
-        goal = FollowJointTrajectoryGoal()
-        goal.trajectory = joint_trajectory
-        self._call_action(goal)
+        goals = {}
+        for controller in self._controllers:
+            controller_joints = self._controllers[controller]
+            goal = FollowJointTrajectoryGoal()
+            goal.trajectory = copy.deepcopy(joint_trajectory)
+
+            indices_of_joints_in_this_controller = []
+
+            for i, joint in enumerate(joint_trajectory.joint_names):
+                if joint in controller_joints:
+                    indices_of_joints_in_this_controller.append(i)
+
+            goal.trajectory.joint_names = [
+                joint_trajectory.joint_names[i] for i in indices_of_joints_in_this_controller]
+
+            for point in goal.trajectory.points:
+                point.positions = [point.positions[i] for i in indices_of_joints_in_this_controller]
```

Comment:
We should also copy point.velocities and point.efforts if present.

Reply:
Good catch. Done